final-project/scripts/blog-carousel.js

```diff
@@ -0,0 +1,173 @@
+const Direction = {
+  NEXT: 1,
+  PREV: -1
+};
+
+export class BlogCarousel {
+  constructor() {
+    this.carouselIndicatorsEl = null;
+    this.carouselItemsEl = null;
+    this.carouselPrevCtrlEl = null;
+    this.carouselNextCtrlEl = null;
+  }
+
+  async init(carouselHostEl) {
+    const carouselInnerEl = document.createElement("div");
+    carouselInnerEl.setAttribute("class", "carousel__inner");
+
+    this.carouselItemsEl = document.createElement("div");
```

Comment:
Probably it could have been easier to manage these HTML nodes by "cloning" them from an existing element in the document, just in case the structure changes, but for this exercise I think is ok.